assets/js/theme/common/aria/checkableOptions.js

```diff
@@ -0,0 +1,61 @@
+import { ariaKeyCodes } from './constants';
+
+const checkItem = (itemCollection, itemIdx) => {
+    itemCollection.each((idx, item) => {
+        const $item = $(item);
+        if (idx !== itemIdx) {
+            $item.attr('aria-checked', false).prop('checked', false);
+            return;
+        }
+
+        $item.attr('aria-checked', true).prop('checked', true).focus();
+    });
+};
+
+const calculateTargetItemPosition = (lastItemIdx, currentIdx) => {
+    switch (true) {
+    case currentIdx > lastItemIdx: return 0;
+    case currentIdx < 0: return lastItemIdx;
+    default: return currentIdx;
+    }
+};
+
+const handleItemKeyDown = (itemCollection, itemIdx) => e => {
+    const { keyCode } = e;
+    const lastCollectionItemIdx = itemCollection.length - 1;
+
+    if (Object.values(ariaKeyCodes).includes(keyCode)) {
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    switch (keyCode) {
+    case ariaKeyCodes.RETURN:
+    case ariaKeyCodes.SPACE: {
+        checkItem(itemCollection, itemIdx);
+        break;
+    }
+    case ariaKeyCodes.LEFT:
+    case ariaKeyCodes.UP: {
+        const prevItemIdx = calculateTargetItemPosition(lastCollectionItemIdx, itemIdx - 1);
+        itemCollection.get(prevItemIdx).focus();
+        break;
+    }
+    case ariaKeyCodes.RIGHT:
+    case ariaKeyCodes.DOWN: {
+        const nextItemIdx = calculateTargetItemPosition(lastCollectionItemIdx, itemIdx + 1);
+        itemCollection.get(nextItemIdx).focus();
+        break;
+    }
+
+    default: break;
+    }
+};
+
+export default ($container, itemSelector) => {
+    const $itemCollection = $container.find(itemSelector);
+
+    $itemCollection.each((idx, item) => {
+        item.addEventListener('keydown', handleItemKeyDown($itemCollection, idx));
+    });
```

Comment:
Smapll proposal - use delegated event handler instead of setting handler on every element
like this
$itemCollection.on('keydown', itemSelector, $itemCollection)
and after that u can figure out index of element inside handler

Reply:
Done